source/MaterialXTest/MaterialXCore/Material.cpp

```diff
@@ -7,9 +7,87 @@
 
 #include <MaterialXCore/Document.h>
 #include <MaterialXCore/Value.h>
+#include <MaterialXFormat/XmlIo.h>
+#include <MaterialXFormat/Util.h>
+
+#include <unordered_set>
 
 namespace mx = MaterialX;
 
+namespace
+{
+
+// Utility function for testing material discovery
+std::vector<mx::InterfaceElementPtr> getMaterials(mx::ElementPtr root, bool skipIncludes)
```

Comment:
I can see where this complex unit test was warranted for the original material discovery logic, but maybe we can now omit this, since the added functionality is a clear-cut collection of getMaterialNodes methods?  I would recommend simplifying this, and simply validating that Document::getMaterialNodes returns the expected set of nodes for your example above.

Reply:
Actually the request was to be able to query from a nodegraph (including nodegraphs for noddefs), a document and materialassign as roots for different workflows. 
Also I think it's better to test all new apis. I'll see about simplifying this however as some cross checks are probably not needed anymore.